### PR TITLE
Reorganize `LLVM_jll`

### PR DESCRIPTION
**Pull request by staticfloat**
We need to provide three separate (but linked) distributions of LLVM:

* `libLLVM_jll`: the base library and include files.  This is what the
Julia distribution will build/link against, and what will ship in the
stdlib for Julia 1.5+, in order to lock LLVM compatibility.

* `clang_jll`: `libclang` and whatnot which is needed for `Clang.jl`.

* `LLVM_jll`: "everything else", such as `opt`, `llc`, etc...

Both `clang_jll` and `LLVM_jll` will depend upon `libLLVM_jll`, and
thereby be forced to match versions with base Julia's LLVM version.

This also adds a utility, `get_addable_spec()` that allows us to _force_ a particular build version of a dependency.  This is useful to ensure that we lock LLVM versions (including patches).

### CODE REVIEW

**Comment by staticfloat** [comment]
@maleadt This is my long-rumored reorg of LLVM.  I have already manually registered [`LLVM_full_jll`](https://github.com/JuliaBinaryWrappers/LLVM_full_jll.jl) (the monolithic build output that gets split into smaller packages), [`libLLVM_jll`](https://github.com/JuliaBinaryWrappers/libLLVM_jll.jl), and [`Clang_jll`](https://github.com/JuliaBinaryWrappers/Clang_jll.jl), but I'm holding off of pushing a new `LLVM_jll` until you OK this, as I don't want to accidentally break your packages that may or may not depend on `LLVM_jll` right now.

File: L/LLVM/Clang@9.0.1/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,5 @@
+name = "Clang"
+libllvm_version = v"9.0.1+2"
```
Comment (vchuravy) [inline L/LLVM/Clang@9.0.1/build_tarballs.jl]: It's a bit annoying that we have to declare that everywhere. Means more places to change it and more places that I can miss.

Comment (staticfloat) [reply L/LLVM/Clang@9.0.1/build_tarballs.jl]: Note that we are building _exactly_ that version; including the build number.  That's why this is important.

File: L/LLVM/common.jl [hunk 2]
```diff
@@ -248,7 +299,6 @@ function configure(ARGS, version)
         "./bundled",
```
Comment (giordano) [inline L/LLVM/common.jl]: Can we use the new types for the sources?

File: L/LLVM/common.jl [hunk 3]
```diff
@@ -262,19 +312,60 @@ function configure(ARGS, version)
...
+    name = "LLVM_full"
     config = "LLVM_MAJ_VER=$(version.major)\n"
     if assert
         config *= "ASSERTS=1\n"
         name = "$(name)_assert"
     end
-    return name, sources, config * buildscript, products
+    # Dependencies that must be installed before this package can be built
+    # TODO: Zlib, LibXML2
+    dependencies = []
```
Comment (giordano) [inline L/LLVM/common.jl]:
```suggestion
    dependencies = Dependency[]
```
to use the new type for the dependencies

File: L/LLVM/Clang@9.0.1/build_tarballs.jl (hunk 1 above)
Comment (giordano) [reply L/LLVM/Clang@9.0.1/build_tarballs.jl]: This information however isn't propagated to the JLL package in the `compat` section though

Comment (staticfloat) [reply L/LLVM/Clang@9.0.1/build_tarballs.jl]: Shoot, you're right.  Grrr.

**Comment by maleadt** [comment]
Awesome!

> I'm holding off of pushing a new LLVM_jll until you OK this, as I don't want to accidentally break your packages that may or may not depend on LLVM_jll right now.

Nothing at this point, as I was holding off working on that LLVM-based tool until these changes. So go ahead, and I'll update my build scripts afterwards. :slightly_smiling_face:

File: L/LLVM/Clang@9.0.1/build_tarballs.jl (hunk 1 above)
Comment (staticfloat) [reply L/LLVM/Clang@9.0.1/build_tarballs.jl]: Okay, fixed, but it won't give as fine-grained control as I want.  We won't be able to control patch versions with this, which means that you may get a newer `clang` or `opt` than the `libLLVM` that shipped with Julia, if we add some patches.  That's _probably_ fine, but not great.

**Pull request merged**